[BUG] types_test.cpp fails to compile from source on Arm/Jetson TX2

**Describe the bug**
Once pre-reqs are met, compiling cuDF from source on a Jetson TX2 via: `./build.sh -v libcudf` yields the following error:

```
[ 94%] Building CXX object tests/CMakeFiles/TYPES_TEST.dir/types/types_test.cpp.o
cd /home/adam/projects/cudf/cpp/build/tests && /usr/bin/c++  -DARROW_METADATA_V4 -I/home/adam/projects/cudf/cpp/build/googletest/install/include -I/usr/local/cuda-10.0/targets/aarch64-linux/include -I/home/adam/projects/cudf/cpp/build/include -I/home/adam/projects/cudf/cpp/include -I/home/adam/projects/cudf/cpp -I/home/adam/projects/cudf/cpp/src -I/home/adam/projects/cudf/cpp/thirdparty/cub -I/home/adam/projects/cudf/cpp/thirdparty/jitify -I/home/adam/projects/cudf/cpp/thirdparty/dlpack/include -I/home/adam/projects/cudf/cpp/build/arrow/install/include -I/home/adam/projects/cudf/cpp/build/arrow/build/flatbuffers_ep-prefix/src/flatbuffers_ep-install/include -I/home/adam/c4aarch64_installer/envs/cudf-dev/include  -Werror -O3 -DNDEBUG -fPIE   -DJITIFY_USE_CACHE -DCUDF_VERSION=0.9.0 -std=gnu++14 -o CMakeFiles/TYPES_TEST.dir/types/types_test.cpp.o -c /home/adam/projects/cudf/cpp/tests/types/types_test.cpp
/home/adam/projects/cudf/cpp/tests/types/types_test.cpp: In member function 'virtual void WrappersTestBool8_BooleanOperatorsBool8_Test::TestBody()':
/home/adam/projects/cudf/cpp/tests/types/types_test.cpp:253:23: error: narrowing conversion of '-42' from 'int' to 'char' inside { } [-Wnarrowing]
     cudf::bool8 w4{-42};
                       ^
/home/adam/projects/cudf/cpp/tests/types/types_test.cpp: In member function 'virtual void WrappersTestBool8_CastArithmeticTest_Test::TestBody()':
/home/adam/projects/cudf/cpp/tests/types/types_test.cpp:286:23: error: narrowing conversion of '-42' from 'int' to 'char' inside { } [-Wnarrowing]
     cudf::bool8 w2{-42};
```

**Steps/Code to reproduce bug**
`./build.sh -v libcudf`
I also disabled a parallel build to better view logs

**Environment details**
```
     ***CPU***
     Architecture:         aarch64
     Byte Order:           Little Endian
     CPU(s):               6
     On-line CPU(s) list:  0,3-5
     Off-line CPU(s) list: 1,2
     Thread(s) per core:   1
     Core(s) per socket:   4
     Socket(s):            1
     Vendor ID:            ARM
     Model:                3
     Model name:           Cortex-A57
     Stepping:             r1p3
     CPU max MHz:          2035.2000
     CPU min MHz:          345.6000
     BogoMIPS:             62.50
     L1d cache:            32K
     L1i cache:            48K
     L2 cache:             2048K
     Flags:                fp asimd evtstrm aes pmull sha1 sha2 crc32
     
     ***CMake***
     /home/adam/c4aarch64_installer/envs/cudf-dev/bin/cmake
     cmake version 3.15.1
     
     CMake suite maintained and supported by Kitware (kitware.com/cmake).
     
     ***g++***
     /usr/bin/g++
     g++ (Ubuntu/Linaro 7.4.0-1ubuntu1~18.04.1) 7.4.0
     Copyright (C) 2017 Free Software Foundation, Inc.
     This is free software; see the source for copying conditions.  There is NO
     warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
     
     
     ***nvcc***
     /usr/local/cuda-10.0/bin/nvcc
     nvcc: NVIDIA (R) Cuda compiler driver
     Copyright (c) 2005-2019 NVIDIA Corporation
     Built on Mon_Mar_11_22:13:24_CDT_2019
     Cuda compilation tools, release 10.0, V10.0.326
     
     ***Python***
     /home/adam/c4aarch64_installer/envs/cudf-dev/bin/python
     Python 3.7.3
     
     ***Environment Variables***
     PATH                            : /home/adam/c4aarch64_installer/envs/cudf-dev/bin:/home/adam/.local/bin:/home/adam/c4aarch64_installer/bin:/usr/local/cuda-10.0/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
     LD_LIBRARY_PATH                 : /usr/local/cuda-10.0/lib64:
     NUMBAPRO_NVVM                   :
     NUMBAPRO_LIBDEVICE              :
     CONDA_PREFIX                    : /home/adam/c4aarch64_installer/envs/cudf-dev
     PYTHON_PATH                     :
     
     ***conda packages***
     /home/adam/c4aarch64_installer/bin/conda
     # packages in environment at /home/adam/c4aarch64_installer/envs/cudf-dev:
     #
     # Name                    Version                   Build  Channel
     boost-cpp                 1.70.0               h8e57a91_2    conda-forge
     bzip2                     1.0.8                h516909a_0    conda-forge
     ca-certificates           2019.6.16            hecc5488_0    conda-forge
     certifi                   2019.6.16                py37_1    conda-forge
     cffi                      1.12.3           py37h8022711_0    conda-forge
     cmake                     3.15.1               hf94ab9c_0    conda-forge
     cmake-setuptools          0.1.3                     <pip>
     curl                      7.65.3               hf8cf82a_0    conda-forge
     cython                    0.29.13          py37he1b5a44_0    conda-forge
     double-conversion         3.1.5                he1b5a44_1    conda-forge
     expat                     2.2.5             h4c5d2ac_1003    conda-forge
     flatbuffers               1.11.0               he1b5a44_0    conda-forge
     icu                       64.2                 h4c5d2ac_0    conda-forge
     krb5                      1.16.3            h6005c1d_1001    conda-forge
     libblas                   3.8.0               12_openblas    conda-forge
     libcblas                  3.8.0               12_openblas    conda-forge
     libcurl                   7.65.3               hda55be3_0    conda-forge
     libedit                   3.1.20170329      hf8c457e_1001    conda-forge
     libffi                    3.2.1             h4c5d2ac_1006    conda-forge
     libgcc-ng                 7.3.0                h5c90dd9_0    c4aarch64
     libgfortran-ng            7.3.0                h6bc79d0_0    c4aarch64
     liblapack                 3.8.0               12_openblas    conda-forge
     libopenblas               0.3.7                h6e990d7_0    conda-forge
     librmm-cffi               0.9.0                     <pip>
     libssh2                   1.8.2                h22169c7_2    conda-forge
     libstdcxx-ng              7.3.0                h5c90dd9_0    c4aarch64
     libuv                     1.30.1               h516909a_0    conda-forge
     llvmlite                  0.29.0                   py37_0    numba
     ncurses                   6.1               hf484d3e_1002    conda-forge
     numba                     0.45.1              np116py37_0    numba
     numpy                     1.16.4           py37h91f3968_0    conda-forge
     nvstrings-cudaunknown     0.0.0.dev0                <pip>
     openssl                   1.1.1c               h516909a_0    conda-forge
     pandas                    0.25.0           py37h59fbc97_0    conda-forge
     pip                       19.2.2                   py37_0    conda-forge
     pycparser                 2.19                       py_0    conda-forge
     python                    3.7.3                habb2978_1    conda-forge
     python-dateutil           2.8.0                      py_0    conda-forge
     pytz                      2019.2                     py_0    conda-forge
     rapidjson                 1.1.0             he1b5a44_1002    conda-forge
     readline                  8.0                  h75b48e3_0    conda-forge
     rhash                     1.3.6             h14c3975_1001    conda-forge
     setuptools                41.0.1                   py37_0    conda-forge
     six                       1.12.0                py37_1000    conda-forge
     sqlite                    3.29.0               h283c62a_0    conda-forge
     tk                        8.6.9             hed695b0_1002    conda-forge
     wheel                     0.33.4                   py37_0    conda-forge
     xz                        5.2.4             hda93590_1001    conda-forge
     zlib                      1.2.11            h516909a_1005    conda-forge
```

**Additional context**
@kkraus14 suggested a work-around:

`cudf::bool8 w4{static_cast<char>(-42)}` and `cudf::bool8 w2{static_cast<char>(-42)}` which enabled the successful compilation of libcudf

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -13,7 +13,7 @@
 ## Bug Fixes
 
 - PR #2584 ORC Reader: fix parsing of `DECIMAL` index positions
-
+- PR #2611 Types Test: fix static casting from negative int to string
 
 # cuDF 0.9.0 (Date TBD)
 

--- a/cpp/tests/types/types_test.cpp
+++ b/cpp/tests/types/types_test.cpp
@@ -250,7 +250,7 @@ TEST_F(WrappersTestBool8, BooleanOperatorsBool8)
     EXPECT_TRUE(w2 >= w3);
     EXPECT_TRUE(w2 <= w3);
 
-    cudf::bool8 w4{-42};
+    cudf::bool8 w4{static_cast<char>(-42)};
     cudf::bool8 w5{43};
 
     EXPECT_TRUE(w4 == w4);
@@ -283,7 +283,7 @@ TEST_F(WrappersTestBool8, BooleanOperatorsBool8)
 TEST_F(WrappersTestBool8, CastArithmeticTest)
 {
     cudf::bool8 w1{42};
-    cudf::bool8 w2{-42};
+    cudf::bool8 w2{static_cast<char>(-42)};
 
     bool t1{42};
     bool t2{-42};

